sleap_io/model/instance.py

```diff
@@ -1,190 +1,266 @@
+        cls, points: np.ndarray, skeleton: Skeleton, track: Optional[Track] = None
+    ) -> "Instance":
+        """Create an instance object from a numpy array.
 
         Args:
-            attribute: Attribute being validated; not used.
-            points: Dict of `points`
-                If dict, keys should be node names.
-
-        Raises:
-            ValueError: If a point is associated with a skeleton node
-                name that doesn't exist.
-            TypeError: With a human readable error message, the attribute (of type
-                attrs.Attribute), the expected type, and the value it got.
+            points: A numpy array of shape `(n_nodes, 2)` corresponding to the points of
+                the skeleton. Values of `np.nan` indicate "missing" nodes.
+            skeleton: The `Skeleton` that this `Instance` is associated with. It should
+                have `n_nodes` nodes.
+            track: An optional `Track` associated with a unique animal/object across
+                frames or videos.
         """
-        if points is not None:
-            try:
-                for node_name in points.keys():
-                    if Node(node_name) not in self.skeleton.nodes:
-                        raise KeyError(
-                            f"There is no node named {node_name} in {self.skeleton}"
-                        )
-            except AttributeError:
-                raise TypeError(
-                    "'{name}' must be {type!r} (got {value!r} that is a "
-                    "{actual!r}).".format(
-                        name=attribute.name,
-                        type=dict,
-                        actual=points.__class__,
-                        value=points,
-                    ),
-                    attribute,
-                    dict,
-                    points,
-                )
+        return cls(
+            points=points, skeleton=skeleton, track=track  # type: ignore[arg-type]
+        )
 
-    @from_predicted.validator
-    def _validate_type_is_PredictedInstance(self, attribute, value):
-        if (value is not None) and (not isinstance(value, PredictedInstance)):
-            raise TypeError(
-                "'{name}' must be {type!r} (got {value!r} that is a "
-                "{actual!r}).".format(
-                    name=attribute.name,
-                    type="PredictedInstance",
-                    actual=value.__class__,
-                    value=value,
-                ),
-                attribute,
-                "PredictedInstance",
-                value,
-            )
+    def numpy(self) -> np.ndarray:
+        """Return the instance points as a numpy array."""
+        pts = np.full((len(self.skeleton), 2), np.nan)
+        for node, point in self.points.items():
+            if point.visible:
+                pts[self.skeleton.index(node)] = point.numpy()
+        return pts
 
 
-@define(auto_attribs=True)
+@define
 class PredictedInstance(Instance):
```

Comment:
I had initially thought we should use explicitly set `auto_attribs=True` in our classes because it forces us to type hint. Is there a reason we decided not to use it?

Reply:
We _do_ type hint everything :)